From f63ae72c8e97c9487a69298f15249b67d39173d6 Mon Sep 17 00:00:00 2001
From: Max Filippov <jcmvbkbc@gmail.com>
Date: Sun, 5 Oct 2014 05:54:31 +0400
Subject: [PATCH 04/29] xtensa: always emit .literal_position for call0 ABI

xtensa assembler implicitly starts literal pool before entry
instructions. Emit .literal_position at the beginning of function for
call0 ABI explicitly, as otherwise direct calls to external symbols
relaxed by assembler will break build when text section literals are
enabled.

Signed-off-by: Max Filippov <jcmvbkbc@gmail.com>
---
 gcc/config/xtensa/xtensa.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcc/config/xtensa/xtensa.h b/gcc/config/xtensa/xtensa.h
index dd315a8..9f4a8a1 100644
--- a/gcc/config/xtensa/xtensa.h
+++ b/gcc/config/xtensa/xtensa.h
@@ -832,7 +832,7 @@ typedef struct xtensa_args
 /* Define output to appear before the constant pool.  */
 #define ASM_OUTPUT_POOL_PROLOGUE(FILE, FUNNAME, FUNDECL, SIZE)          \
   do {									\
-    if ((SIZE) > 0)							\
+    if ((SIZE) > 0 || !TARGET_WINDOWED_ABI)				\
       {									\
 	resolve_unique_section ((FUNDECL), 0, flag_function_sections);	\
 	switch_to_section (function_section (FUNDECL));			\
-- 
2.1.4

